In [6]:
import numpy as np
import scipy.stats as stats
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
n = len(data)
sample_mean = np.mean(data)
sample_std = np.std(data, ddof=1) 

In [7]:
sample_mean

1.2386666666666666

In [8]:
sample_std

0.19316412956959936

In [9]:
standard_error_t = sample_std / np.sqrt(n)
task_a_interval = stats.t.interval(0.99, df=n-1, loc=sample_mean, scale=standard_error_t)

print("--- TASK A OUTPUT ---")
print("Confidence Interval (T-dist):", task_a_interval)

--- TASK A OUTPUT ---
Confidence Interval (T-dist): (1.0901973384384906, 1.3871359948948425)


In [10]:
pop_std_dev = 0.2  
standard_error_z = pop_std_dev / np.sqrt(n)
task_b_interval = stats.norm.interval(0.99, loc=sample_mean, scale=standard_error_z)

print("--- TASK B OUTPUT ---")
print("Confidence Interval (Z-dist):", task_b_interval)

--- TASK B OUTPUT ---
Confidence Interval (Z-dist): (1.1056514133957607, 1.3716819199375725)


### Explaination of task A

Rationale for using T-Distribution:
In this scenario, we are tasked with constructing a 99% confidence interval using the sample standard deviation. We are testing a small sample of 15 print-heads due to the destructive nature of the testing process. Because our sample size is small ($n < 30$) and the actual Population Standard Deviation is completely unknown to us, using a standard Normal (Z) distribution might lead to inaccurate estimations. Therefore, we use the T-distribution. The T-distribution is slightly wider and correctly accounts for the extra risk and uncertainty associated with small sample sizes. 

### Explaination of task B

In this task, the condition changes. We are given that the population standard deviation is 0.2 million characters. Whenever the exact Population Standard Deviation is known, the uncertainty regarding the spread of the data is eliminated. In such cases, the rule of statistics dictates that we no longer need the T-distribution. We can directly apply the Z-distribution (Normal distribution) to calculate the 99% confidence interval for the mean number of characters printed before failure

### Interpretation / Conclusion (Task A):

Based on the T-distribution calculation, the 99% confidence interval is approximately [1.090, 1.387].
This means: We can be 99% confident that the true population mean of the print-head durability lies between 1.090 million and 1.387 million characters. Because we relied purely on a small sample of 15 units without knowing the actual population variance, this interval is slightly wider to safely account for the statistical uncertainty. From a business perspective, the manufacturer can confidently guarantee that their print-heads will, on average, survive at least ~1.09 million characters before failing.

### Interpretation / Conclusion (Task B):

Based on the Z-distribution calculation, the 99% confidence interval is approximately [1.105, 1.371].
What this means: We can be 99% confident that the true population mean lies between 1.105 million and 1.371 million characters.
Observation (A vs B): When we compare this to Task A, we see that this interval is narrower (more precise). This happens because knowing the exact population standard deviation (0.2 million) removes a layer of guesswork and risk. The manufacturer now has a tighter, more accurate estimation window for their product's durability.